shm/inc/ship.hpp

```diff
@@ -25,6 +26,7 @@ class Ship {
     std::string getName() const { return name_; }
     size_t getId() const { return id_; }
     Cargo getCargo(size_t index) const;
+    std::vector<Cargo>& getallCargo() { return cargo_; }

```

Comment:
Do we really need to make our whole cargo vector effectively public in Ship?
Idea:
In Ship a public function, that returns the available space (use solution @kontrybutor suggested).
In Player - just call that function.